sw/device/tests/sim_dv/pwrmgr_random_sleep_all_wake_ups.c

```diff
@@ -70,36 +68,27 @@ bool test_main(void) {
 
   // Enable pwrmgr interrupt.
   CHECK_DIF_OK(dif_pwrmgr_irq_set_enabled(&pwrmgr, 0, kDifToggleEnabled));
-
-  // Enable access to flash for storing info across resets.
-  LOG_INFO("Setting default region accesses");
-  CHECK_STATUS_OK(
-      flash_ctrl_testutils_default_region_access(&flash_ctrl,
-                                                 /*rd_en*/ true,
-                                                 /*prog_en*/ true,
-                                                 /*erase_en*/ true,
-                                                 /*scramble_en*/ false,
-                                                 /*ecc_en*/ false,
-                                                 /*he_en*/ false));
-
   uint32_t wakeup_count = 0;
-  CHECK_STATUS_OK(flash_ctrl_testutils_counter_get(0, &wakeup_count));
-  uint32_t wakeup_unit = get_wakeup_unit(wakeup_count);
-  bool deep_sleep = get_deep_sleep(wakeup_count);
+  uint32_t wakeup_unit = 0;
+  bool deep_sleep = false;
 
   if (UNWRAP(pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) == true) {
     LOG_INFO("POR reset");
     execute_test(wakeup_unit, deep_sleep);
     check_wakeup_reason(wakeup_unit);
     LOG_INFO("Woke up by source %d", wakeup_unit);
     cleanup(wakeup_unit);
-    CHECK_STATUS_OK(flash_ctrl_testutils_counter_increment(&flash_ctrl, 0));
-    CHECK_STATUS_OK(flash_ctrl_testutils_counter_get(0, &wakeup_count));
+    CHECK_STATUS_OK(ret_sram_testutils_counter_clear(0));
```

Comment:
Nit: I wonder whether it would make sense to stick the counter index in a preprocessor macro or some `const size_t counter_id = 0`?

I only ask because I didn't know the API already and had to look up why it looked like we were incrementing by zero :-)

Reply:
This always increments by 1. The zero is the counter index: there are 4 counters, so [0..3] are the legal indices. In many uses of these counters we give them local names with an enum. I just noticed ret_sram_testutils.h should export kNumberOfCounters as well, and "mangle" it as kRetSramTestutilsNumberOfCounters. By the way, the current code is no worse than the flash_ctrl counters.

I'd prefer to defer these changes since I am making more changes to this file, and I have already added an enum for the counter, plus a few more changes.

Reply:
Thanks for explaining the code: I had indeed understood it, but it took me a few seconds and I was suggesting a name that might make things more obvious for the next reader :-)

Anyway, if you've got a chain of changes and don't want to rebase them, I totally understand wanting to do this as a followup.